### DISCUSSION TITLE
What is Microsoft.Maui.Graphics.Native?

### DISCUSSION THREAD

**Discussion by mcguu** [dotnet/maui#4959] [Q&A]
where to find it?

I can not find the class NativeGraphicsView in Microsoft.Maui.Graphics.Controls

**Comment by jsuarezruiz** [comment]
That class has been renamed to **PlatformGraphicsView**.

**Comment by mcguu** [comment]
@jsuarezruiz Thank you for your answer.

There is another question need I ask for a new post?

I want to make a custom view : MyView1

```
public interface IMyView1:GraphicsView
{

}

public class MyView1:IMyView1
{
public MyView1()
{
     Drawable=new MyDrawable();
}
}
```

and I want to do android code in **Platform/android** and ios in ### Platform/ios

SO I Should use same name in there ?

```
public partial class MyView1:Android.Views.View
{
     public MyView1(Context context):base(context){}//there will be error.
}
```

Sorry for disturb you and you have so many issue to deal but I am a newer to maui and do not know where to find it.

**Reply by pictos** [reply]
Yeah, same name and same `namespace`. Also, if you aren't building a nuget package you can skip the interface part (;

**Reply by mcguu** [reply]
But there will be error can not find suitable constructor there will be different constructor in ios and android and the view right?

**Reply by pictos** [reply]
Do you have a repo that I can take a look at? In the meantime you can see [this talk](https://youtu.be/zWiZlafN5EI) that I gave about this subject

**Reply by mcguu** [reply]
Now I test and no error know I do not know why.

thank you @pictos

can you please tell me how to register a handler? and I find that If I do not register it it will run ok too.

**Reply by mcguu** [reply]
[repo.zip](https://github.com/dotnet/maui/files/8166114/repo.zip)

Now I find that in the class (partial) I can not use a method that in another same name class?

int the Platform/android/PoView.cs I can not use the method in PoView.cs.
@pictos

**Reply by mcguu** [reply]
and I test the PoView.cs in Platform/android is not called.I do not know why?

**Reply by pictos** [reply]
@mcguu I'll take a look by tomorrow. If you have a method on `PoView.cs` it will be compiled for all platforms, that said you can't have the same method name in both `shared` and `platform` files. Maybe a partial method could do the job?

**Reply by mcguu** [reply]
@pictos  thank you.

I do not have a same method name in both **shared** and **platform** but only in shared and want to be called in platform.

please sleep now and try it when you have some time.

**Reply by mcguu** [reply]
@pictos  any updates?

**Reply by pictos** [reply]
@mcguu so, the issue is because of the namespace... You `PoView.cs` has `namespace repo` and the `PoView.cs`, inside the Android folder, has `namespace repo.Platforms.Android`. Both should be in the same namespace to be the same class. When you changed that, you will see some compiler errors saying that `partial class` can't inherent from different a base class

**Reply by mcguu** [reply]
@pictos  so what should I do to this error of the partial class can't inherit from different base class?

or let me say how to create a custome view? in the platform of android?

**Reply by pictos** [reply]
I don't what you want to do. But I can see some solutions 
- Create a handler
- you inherit from Drawing View, I believe that view has a Handler and Native View, so you can take advantage of that and implement what you need using just a custom view 
- create some attached properties and helper methods per platform and use them to extend the control the way that you want

**Comment by mcguu** [comment]
@pictos What I want to do is to know how to make a custom view.

the view how to connect with what in platform/android and platform/ios.

now the PoView in platform/android can not be called I do not know why.

thank you @pictos  It is time for you to sleep right?

**Reply by pictos** [reply]
Yeah, it was hehe. Well see if this can give you some insight
- https://github.com/pictos/MauiSpecRepo/blob/master/MauiSpecRepo/CustomControl/ImageTintColor.cs
